### DISCUSSION TITLE
What feature do you need?

### DISCUSSION THREAD

**Discussion by south1907** [south1907/addmember-telegram#72] [Ideas]
Don't hesitate to show me what you want

**Comment by south1907** [comment]
Todo: group_source can add by group id.
Current is only group username, so some group don't have group username

**Reply by NginxForbiden** [reply]
Thanks. Im patiently waiting for this feature.

**Comment by south1907** [comment]
Todo: create video turorial to help someone run Project.
--> DONE

**Reply by south1907** [reply]
Done: https://www.youtube.com/watch?v=qnNsgHCMQYk
Use my local voice, but have transcript English

**Comment by south1907** [comment]
Todo: Check member in target group before run
--> DONE

**Reply by south1907** [reply]
I have recently update logic: check member exists in target group, if it exists, not re-add to target:
commit: https://github.com/south1907/addmember-telegram/commit/28553a42a0a84f2fcc8df686cac0646f19cbe441

**Reply by nguyeenxtrang10** [reply]
Greate job!

**Comment by south1907** [comment]
Todo: Add member multiple (not complete because check error flood when run)
![add-multiple-in-one](https://user-images.githubusercontent.com/24355759/207102195-e161760c-f37a-4525-bc4e-4e9472ce5f84.png)

**Comment by south1907** [comment]
Todo: Trace log added and check Privacy while add
--> DONE

**Comment by tsoodo** [comment]
system to prevent perl flood error. i loaded it up with the recommended 20 accs and i got rate limited

**Reply by south1907** [reply]
And it is still flood?

**Comment by Vikkiing** [comment]
Is it currently possible to pull from groups without a username? (google translator)

**Reply by south1907** [reply]
Yes, it is still working

**Comment by ofekb** [comment]
It is possible to add users from an Excel or CSV file as a name source? There is a list of usernames

**Reply by south1907** [reply]
Yes, but require username.
I will add it to project
https://github.com/users/south1907/projects/2/views/1?pane=issue&itemId=19898160

**Comment by isaaccwb** [comment]
Hello, thanks for your video
I have Two questions

1:  can I add sleep time to get data , i have so many groups I don’t want to get reach to TG limit

Also when I run the telethon it will stop to scrape further groups around 100 , do we need to optimize the session time or chunk size says I have 2000 groups 

2: can I just add specific member in the add member process ?

Thanks
